crates/eips/src/eip2718.rs

```diff
@@ -2,28 +2,46 @@
 //!
 //! [EIP-2718]: https://eips.ethereum.org/EIPS/eip-2718
 
+use alloc::{vec, vec::Vec};
 use alloy_primitives::{keccak256, Sealed, B256};
 use alloy_rlp::{BufMut, Header, EMPTY_STRING_CODE};
+use core::{
+    fmt,
+    fmt::{Display, Formatter},
+};
 
 // https://eips.ethereum.org/EIPS/eip-2718#transactiontype-only-goes-up-to-0x7f
 const TX_TYPE_BYTE_MAX: u8 = 0x7f;
 
 /// [EIP-2718] decoding errors.
 ///
 /// [EIP-2718]: https://eips.ethereum.org/EIPS/eip-2718
-#[derive(thiserror::Error, Debug)]
+#[derive(Debug, Copy, Clone)]
 pub enum Eip2718Error {
     /// Rlp error from [`alloy_rlp`].
-    #[error(transparent)]
-    RlpError(#[from] alloy_rlp::Error),
+    RlpError(alloy_rlp::Error),
     /// Got an unexpected type flag while decoding.
-    #[error("Unexpected type flag. Got {0}.")]
     UnexpectedType(u8),
-    /// Some other error occurred.
-    #[error(transparent)]
-    Custom(#[from] Box<dyn std::error::Error>),
```

Comment:
I think eventually we'd need this, so perhaps the better question for this PR is, does this type even belong in here? 

perhaps we should rather move this to the consensus crate instead, I believe it's better suited in there

Reply:
What do you mean?
Move all eip error type to consensus crate? and add Error type to trait such as `Decodable2718`?

Reply:
But that might be a break change, which the current one is not?

Reply:
removing a variant is a breaking change as well, I'd really prefer moving the error. imo it does not belong in the eip crate